test/test.dart

```diff
@@ -710,3 +710,16 @@ Future<void> enableAllAutomaticRecovery() async {
     await client.setAutomaticRecoveryEnabled(appName, true);
   }
 }
+
+
+extension $WriteExtension on Realm {
```

Comment:
Why do we need this for all tests. I think this is very specific to the tests we are changing in this PR so I would suggest moving this close to where it is used. I don't think we are going to use these in any other place.

Reply:
also making this extension private after the move

Reply:
I would also consider not using an extension since it is visible in the whole file and this is not useful in that file as well except the test that are using it. I would move this to a regular method